bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcarePetHandler.java

```diff
@@ -0,0 +1,251 @@
+
+    private static final String JPEG_CONTENT_TYPE = "image/jpeg";
+
+    private static final ByteArrayFileCache IMAGE_CACHE = new ByteArrayFileCache("org.openhab.binding.surepetcare");
+
+    public SurePetcarePetHandler(Thing thing, SurePetcareAPIHelper petcareAPI) {
+        super(thing, petcareAPI);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateThingCache.getValue();
+        } else {
+            switch (channelUID.getId()) {
+                case PET_CHANNEL_LOCATION:
+                    logger.debug("Received location update command: {}", command.toFullString());
+                    if (command instanceof StringType) {
+                        synchronized (petcareAPI) {
+                            SurePetcarePet pet = petcareAPI.getPet(thing.getUID().getId());
+                            if (pet != null) {
+                                String newLocationIdStr = ((StringType) command).toString();
+                                Integer newLocationId = Integer.valueOf(newLocationIdStr);
+                                // Only update if location has changed. (Needed for Group:Switch item)
+                                if ((pet.status.activity.where.equals(newLocationId)) || newLocationId.equals(0)) {
+                                    logger.debug("Location has not changed, skip pet id: {} with loc id: {}", pet.id,
+                                            newLocationId);
+                                } else {
+                                    try {
+                                        logger.debug("Received new location: {}", newLocationId);
+                                        petcareAPI.setPetLocation(pet, newLocationId, new Date());
+                                        updateState(PET_CHANNEL_LOCATION,
+                                                new StringType(pet.status.activity.where.toString()));
+                                        updateState(PET_CHANNEL_LOCATION_CHANGED,
+                                                new DateTimeType(pet.status.activity.getLocationChanged()));
+                                    } catch (NumberFormatException e) {
+                                        logger.warn("Invalid location id: {}, ignoring command", newLocationIdStr);
+                                    } catch (SurePetcareApiException e) {
+                                        logger.warn("Error from SurePetcare API. Can't update location {} for pet {}",
+                                                newLocationIdStr, pet);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    break;
+                case PET_CHANNEL_LOCATION_TIMEOFFSET:
+                    logger.debug("Received location time offset update command: {}", command.toFullString());
```

Comment:
Consider removing the call to `toFullString` the implemenation of toString is not much different and with the current code  the command `command.toFullString()` will always be executed even when debug logging is not enabled.

Reply:
Apply this also to other applicable locations

Reply:
done. changed in 2 locations